fabric-api-base/src/main/java/net/fabricmc/fabric/impl/base/event/ArrayBackedEvent.java

```diff
@@ -50,21 +53,27 @@ void update() {
 		}
 	}
 
+	@SuppressWarnings("unchecked")
 	@Override
 	public void register(T listener) {
 		if (listener == null) {
 			throw new NullPointerException("Tried to register a null listener!");
 		}
 
-		if (handlers == null) {
-			//noinspection unchecked
-			handlers = (T[]) Array.newInstance(type, 1);
-			handlers[0] = listener;
-		} else {
-			handlers = Arrays.copyOf(handlers, handlers.length + 1);
-			handlers[handlers.length - 1] = listener;
-		}
+		lock.lock();
```

Comment:
We can just use `synchronized` as we don't really need fancy stuff from the lock.

From https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/util/concurrent/locks/ReentrantLock.html:
> ... with the same basic behavior and semantics as the implicit monitor lock accessed using synchronized methods and statements

Reply:
Player wanted to avoid associating a monitor with the object in order not to influence the lock-free path.